packaging/tags.py

```diff
@@ -539,20 +547,19 @@ def _parse_glibc_version(version_str):
     return (int(m.group("major")), int(m.group("minor")))
 
 
-def _have_compatible_glibc(tag_major, tag_minor):
-    # type: (int, int) -> bool
-    version_str = _glibc_version_string()
-    if version_str is None:
-        return False
-    return _check_glibc_version(version_str, tag_major, tag_minor)
+_glibc_version = []  #  type: List[Tuple[int, int]]
 
 
 def _get_glibc_version():
     # type: () -> Tuple[int, int]
+    if _glibc_version:
+        return _glibc_version[0]
     version_str = _glibc_version_string()
     if version_str is None:
-        return -1, -1
-    return _parse_glibc_version(version_str)
+        _glibc_version.append((-1, -1))
+    else:
+        _glibc_version.append(_parse_glibc_version(version_str))
+    return _glibc_version[0]
```

Comment:
use a pseudo-lru_cache to cache the return value from the relatively expensive system call